src/app/activity/bounty/Submit.ts

```diff
@@ -1,45 +1,130 @@
-import Log from '../../utils/Log';
-import { GuildMember } from 'discord.js';
-import MongoDbUtils from '../../utils/MongoDbUtils';
 import mongo, { Db, UpdateWriteOpResult } from 'mongodb';
-import { BountyCollection } from '../../types/bounty/BountyCollection';
-import { CustomerCollection } from '../../types/bounty/CustomerCollection';
-import { BountyStatus } from '../../constants/bountyStatus';
-import BountyUtils from '../../utils/BountyUtils';
-
+import MongoDbUtils from "../../utils/MongoDbUtils";
+import Log from "../../utils/Log";
+import BountyUtils from "../../utils/BountyUtils";
+import DiscordUtils from "../../utils/DiscordUtils";
+import { ButtonInteraction, GuildMember, ModalOptions, ModalSubmitInteraction } from "discord.js";
+import { ComponentType, TextInputStyle, ModalInteractionContext, ModalOptions as scModalOptions, CommandContext } from "slash-create";
+import RuntimeError from "../../errors/RuntimeError";
+import ValidationError from "../../errors/ValidationError";
+import ModalTimeoutError from "../../errors/ModalTimeoutError";
+import { SubmitRequest } from "../../requests/SubmitRequest";
+import { BountyCollection } from "../../types/bounty/BountyCollection";
+import { BountyStatus } from "../../constants/bountyStatus";
+import { CustomerCollection } from "../../types/bounty/CustomerCollection";
 
 export const submitBounty = async (request: SubmitRequest): Promise<void> => {
-	Log.debug('In Submit activity');
-	
-    const getDbResult: {dbBountyResult: BountyCollection, bountyChannel: string} = await getDbHandler(request);
-	// Since card may have been in a DM, guild might not be populated in the request
-	if (request.guildId === undefined || request.guildId === null) {
-		request.guildId = getDbResult.dbBountyResult.customerId;
-	}
-    const submittedByUser = await DiscordUtils.getGuildMemberFromUserId(request.userId, request.guildId);
-	const createdByUser: GuildMember = await submittedByUser.guild.members.fetch(getDbResult.dbBountyResult.createdBy.discordId);
-	Log.info(`${request.bountyId} bounty submitted by ${submittedByUser.user.tag}`);
-
-    await writeDbHandler(request, submittedByUser);
-
-    const cardMessage = await BountyUtils.canonicalCard(getDbResult.dbBountyResult._id, request.activity);
-
-	let creatorSubmitDM = `Please reach out to <@${submittedByUser.user.id}>. They are ready for bounty review`
-
-	if (request.url) {
-		creatorSubmitDM += `\nPlease review this URL:\n${request.url}`
-	}
-
-	if (request.notes) {
-		creatorSubmitDM += `\nPlease review these notes:\n${request.notes}`
-	}
-	await DiscordUtils.activityNotification(creatorSubmitDM, createdByUser, cardMessage.url);
-	await DiscordUtils.activityResponse(request.commandContext, request.buttonInteraction, `Bounty in review! Expect a message from <@${createdByUser.id}>.`, cardMessage.url);
-    return;
-}
+    Log.debug('In Submit activity');
+
+    // Different modal data types and calls in slash commands vs. button interactions
```

Comment:
Thanks for calling this out - why is this the case? 

Where I'm coming from: in the past we invested into making bounty activities as agnostic as possible to the interaction that kicked them off (from the web, from a slash command, from a button interaction). So seeing this logic here begs the question of why break that pattern? 

(and if it has to do with modal properties differing between slash commands and buttons, then this approach is fine - I'd just like to understand what it is)

Reply:
It's because slash-create doesn't support the standard interaction properties needed to make the other one work (I'd have to dig back into my notes to remember why). So either we use the 2 patterns or refactor slash-create out which I didn't want to take on.